schema/elements.go

```diff
@@ -259,3 +259,24 @@ func (s *Schema) Resolve(tr TypeRef) (Atom, bool) {
 	}
 	return tr.Inlined, true
 }
+
+// Clones this isntance of Schema into the other
+// If other is nil this method does nothing.
+// If other is already initialized, overwrites it with this instance
+func (s Schema) CopyInto(dst *Schema) {
+	if dst == nil {
+		return
+	}
+
+	// Schema type is considered immutable so sharing references
+	dst.Types = s.Types
+
+	if s.m != nil {
```

Comment:
Can you check `s.m` like this? What if the once is running on that type somewhere else?

Reply:
My intent was not to make this thread-safe with respect to `dest`. But you are right it should be thread-safe with respect to `s`. Will fix

Reply:
My intent was not to make this thread-safe with respect to `dest`. But you are right it should be thread-safe with respect to `s`. Will fix

Reply:
@apelisse  try again now